### DISCUSSION TITLE
Alpine pattern for server side components?

### DISCUSSION THREAD

**Discussion by galaczi** [alpinejs/alpine#2896] [1. Help]
I am quite new to Alpine and wondering if there is a pattern for creating Alpine-powered components. For example I have multiple custom multiselect components created server side that would require their own separate state client-side. So far I have created them by using Alpine.data in a script tag inside the server side component. With a larger component that means that I would duplicate the content of the whole Alpine.data setup in the script tag of each component inside html. Virtually everything is the same other than Alpine.data's name and a couple settings.

Is there a pattern people use to avoid this kind of duplication (and thus reducing html size)?

**Comment by philwolstenholme** [comment]
I think I'd have a single `<script>` element that uses `Alpine.data` and then I'd [pass in the name and settings as parameters](https://alpinejs.dev/globals/alpine-data#initial-parameters).
